enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,472 @@
+> starting state and then list the steps that the user would need to go through to
+> trigger the feature described in the enhancement. Optionally add a
+> [mermaid](https://github.com/mermaid-js/mermaid#readme) sequence diagram.
+> 
+> Use sub-sections to explain variations, such as for error handling,
+> failure recovery, or alternative outcomes.
+
+**cluster-admin** is a human user responsible for managing a cluster.
+
+1. Start with a 4.18 cluster with conflicting CRDs.
+2. Upgrade to 4.19.
+3. Check clusteroperators, see a conflict.
+4. Run some `oc` command.
+5. Check the ingress clusteroperator again.  Now everything should be dandy.
+
+_TBD: Fill in the details for handing ownership of CRD life-cycle management to
+the Ingress Operator in the case of a conflict, using Server-Side Apply._
+
+### API Extensions
+
+None.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Hypershift runs the Ingress Operator on the management cluster but configured
+with a kubeconfig to manage resources on the guest cluster.  This means that the
+Ingress Operator manages the Gateway API CRDs on the guest cluster, the same as
+on standalone clusters.
+
+#### Standalone Clusters
+
+For standalone clusters, the Ingress Operator manages the Gateway API CRD
+life-cycle so that Gateway API can be configured and used post-install.
+
+#### Single-node Deployments or MicroShift
+
+The CRDs themselves use minimal resources.  Creating a GatewayClass CR can cause
+the Ingress Operator to install OpenShift Service Mesh, which in turn installs
+Istio and Envoy (see the [gateway-api-with-cluster-ingress-operator](gateway-api-with-cluster-ingress-operator.md)
+enhancement proposal), which use considerable resources.  For Single-Node
+OpenShift, the cluster-admin might be advised to pay particular attention to
+OSSM's resource requirements and the cluster's resource constraints before
+attempting to use Gateway API.
+
+MicroShift does not run the Ingress Operator and has its own design for
+supporting Gateway API (see the MicroShift [gateway-api-support](../microshift/gateway-api-support.md)
+enhancement).
+
+### Implementation Details/Notes/Constraints
+
+> What are some important details that didn't come across above in the
+> **Proposal**? Go in to as much detail as necessary here. This might be
+> a good place to talk about core concepts and how they relate. While it is useful
+> to go into the details of the code changes required, it is not necessary to show
+> how the code will be rewritten in the enhancement.
+
+### Risks and Mitigations
+
+#### Dead fields
+
+If an installed Gateway API CRD includes some field that is not implemented by
+the installed Gateway API implementation, then updates to this field may be
+silently ignored.  In the context of a security feature, ignoring this field
+could result in a configuration that inadvertently exposes workload.
+
+Potential mitigation strategies:
+
+- Handle with an admission webhook.
```

Comment:
(just a note that we put handling updates to dead fields being handled by a webhook as a non-goal above)

We have a dilemma when it comes to trying to handle dead fields with a webhook, because our validation would require us to ultimately reference other objects. Something we need to think through, but we can follow up on that as part of the TBD.

Reply:
I added a cross-reference to the "Admission webhook" section under "Alternatives", and I added a paragraph about consistency issues to that section.  I left the TBD to fill in more details.  These changes are in https://github.com/openshift/enhancements/compare/99aae59bb1ac50f1ef02e07ff3dcc3385216c451..e34441f0d0685989e6417db70dcf049c68f9fc55.